### PR TITLE
Convert Windurst Missions to Interaction Framework

### PR DESCRIPTION
**Pull request by claywar**
**General**
- Was able to verify Windurst Walls nearby NPCs default dialogue prior to immigrating

**Windurst 1-1 : The Horutoto Ruins Experiment** 
- Merged PR for this one is here: #498 
- Had the opportunity to run back through and verify some of 1-1 as well as do some off-repo style requests.
- Per recent discussion with some folks, deleted the now useless Windy 1-1 Magical Gizmo NPC scripts.
- Encountered instance of a likely unintentional CS with Mokyokyo hitting the cutting room floor and not following the typical pattern. I have made a judgement call and restored and notated this. If reviewers don't like it, I can throw in the retail event and leave the lost content in a comment.

**Windurst 1-2 : The Heart of the Matter** 
 - Confirmed these Magical Gizmos have no DefaultAction outside of the mission, via checking after mission is completed and on a 2nd non-Windurst citizen character. Uncertain about prior to receiving the mana orbs from Apururu at the start of the mission. msgBase offset +12 reads "This appears to be a device that recharges Mana Orbs." Possibly unused? We originally had it displayed outside of the mission.
 - Confirmed where KIs were silently deleted.
 - Verified both possible endings of this quest thanks to Wiggo's capture of the orbs being lost and warping on my own character to capture the outcome of saving them. 

**Windurst 1-3: The Price of Peace**
 - Confirmed Yagudo NPCs' dialogue during and after the mission, for a Windurstian and Bastokan (who did previously complete the mission before immigrating back to Bastok). Also tested for double and triple tapping. Did not find an instance of Ghoo's event 51 or Laa's 46 triggering, so presuming it's cutting room floor dialogue. 
 - Confirmed new dialogue for Kenapa-Keppa.
 - Correct keyitem names for DRINK_OFFERING and FOOD_OFFERING to match in-game.

**Windurst 2-1: Lost for Words**
 - Corrected message text per retail capture for Fossil Rocks interactions.
 - Added/confirmed optional dialogue for Nanaa's cronies and the NPCs in the same room as Tosuka-Porika at various stages of the mission.

**Windurst 2-2: A Testing Time**
 - Changed the check for the mission being active from every mob killed in game to just non-NMs in Tahrongi Canyon and Buburimu Peninsula. This now occurs in zone-specific globals and is called from onMobDeath from just those mobs.
 - Added/confirmed optional dialogue for many school children and masters.
 - Was unable to 100% verify the killCounts needed for every CS. I kept the existing values in place.
 - Need to verify gate guard events post-mission, if any exist.

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Incorporates Lusi's Windurst branch into additional work for later missions.  This is a mission to eliminate finishMissionTimeline once and for all.

### CODE REVIEW

**Comment by claywar** [comment]
Opening this up.  Still need to run through and test everything, but initial work is complete.

Edit: Need to update mission script comments as well to reflect relevant NPCs

**Comment by claywar** [comment]
Tested through all, should be good to go.

File: scripts/missions/windurst/1_1_The_Horutoto_Ruins_Experiment.lua [hunk 1]
```diff
@@ -319,8 +316,8 @@ mission.sections =
 
         [xi.zone.PORT_WINDURST] =
         {
-            ['Hakkuru-Rinkuru'] = mission:progressEvent(94, 0, xi.ki.CRACKED_MANA_ORB),
-            ['Kuroido-Moido'] = mission:event(98),
+            ['Hakkuru-Rinkuru'] = mission:progressEvent(94, 0, xi.ki.CRACKED_MANA_ORBS),
```
Comment (zach2good) [inline scripts/missions/windurst/1_1_The_Horutoto_Ruins_Experiment.lua]: Should this be `CRACKED_MANA_ORB`?

File: scripts/missions/windurst/1_2_The_Heart_of_the_Matter.lua [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+        else
+            --orb has already been retrieved
+            player:messageSpecial(msgBase + 3) -- "You have already retrieved a glowing Mana Orb from here."
+        end
+    end
+end
+
+local placeOrb = function(player, csid, option, npc)
+    local gizmoNum = npc:getID() - outerHorutotoRuinsID.npc.GATE_MAGICAL_GIZMO -- gizmoNum will be 1 through 6
+    local numberPlaced = mission:getVar(player,"OrbsPlaced")
```
Comment (zach2good) [inline scripts/missions/windurst/1_2_The_Heart_of_the_Matter.lua]: Space between args please

File: scripts/missions/windurst/1_2_The_Heart_of_the_Matter.lua [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+                    player:setMissionStatus(mission.areaId, 3)
+                end,
+            },
+        },
+    },
+
+    -- Handle both the placing and later collecting of mana orbs here
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 3 or missionStatus == 4
```
Comment (zach2good) [inline scripts/missions/windurst/1_2_The_Heart_of_the_Matter.lua]: Not entirely clear the order of operations you want here, please add parens to clarify

File: scripts/missions/windurst/1_3_The_Price_of_Peace.lua [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+
+    -- Finish mission with a gate guard
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 4
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Janshura-Rashura'] =  mission:progressEvent(114),
```
Comment (zach2good) [inline scripts/missions/windurst/1_3_The_Price_of_Peace.lua]: Theres a bunch of funky styling around here, you can find it by searching for "=  " (equals space space)

File: scripts/missions/windurst/2_1_Lost_for_Words.lua [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+
+    if rockOffset == correctRock then
+        if missionStatus == 3 then
+            player:setMissionStatus(mission.areaId, 4)
+            return mission:keyItem(xi.ki.LAPIS_CORAL)
+        elseif missionStatus == 4 then
+            return mission:messageSpecial(mazeID.text.NO_NEED_INVESTIGATE)
+        end
+    else
+        -- TODO: This multiline message should be converted to messageName, however the equivalent in
```
Comment (zach2good) [inline scripts/missions/windurst/2_1_Lost_for_Words.lua:55]: Before this PR is merged, can you turn any outstanding TODOS into issues that contain links to the offending lines so we can track them better

**Review by zach2good — COMMENTED** [review]
Reviewed up to 2-3-1, looking good so far 👍

**Review by Xaver-DaRed — APPROVED** [review]
Tested and got to the end with no major odities.

**Review by zach2good — APPROVED** [review]

**Pull request merged**